calculator.js

```diff
@@ -1,5 +1,223 @@
+/// BASE CASE SAMPLE INPUTS ///
+
 const exampleAdditionInput = {
   num1: 3,
   num2: 5,
-  operation: 'add',
+  op: 'Add',
 }
+
+const exampleAdditionInputSign = {
+  num1: 3,
+  num2: 5,
+  op: '+',
+}
+
+const exampleSubtractInput = {
+  num1: 10,
+  num2: 5,
+  op: 'subtract',
+}
+
+const exampleSubtractInputSign = {
+  num1: 10,
+  num2: 5,
+  op: '-',
+}
+
+const exampleMultiInput = {
+  num1: 3,
+  num2: 5,
+  op: 'multiply',
+}
+
+const exampleMultiInputSign = {
+  num1: 3,
+  num2: 5,
+  op: '*',
+}
+
+const exampleDivideInput = {
+  num1: 3,
+  num2: 5,
+  op: 'divide',
+}
+
+const exampleDivideInputSign = {
+  num1: 3,
+  num2: 5,
+  op: '/',
+}
+
+const exampleModuloInput = {
+  num1: 5,
+  num2: 3,
+  op: 'modulo',
+}
+
+const exampleModuloInputSign = {
+  num1: 5,
+  num2: 3,
+  op: '%',
+}
+
+const exampleExponentInput = {
+  num1: 2,
+  num2: 3,
+  op: 'exponent',
+}
+
+const exampleExponentInputSign = {
+  num1: 2,
+  num2: 3,
+  op: '**',
+}
+
+/// EDGE CASE SAMPLE INPUTS ///
+
+const exampleMissingNum = {
+  num1: null,
+  num2: 5,
+  op: '+',
+}
+
+const exampleMissingOp = {
+  num1: 3,
+  num2: 5,
+  op: null,
+}
+
+const exampleWrongSign = {
+  num1: 3,
+  num2: 5,
+  op: '(',
+}
+
+const exampleOneNonNum = {
+  num1: 'hotdog',
+  num2: 5,
+  op: '/',
+}
+
+const exampleTwoNonNum = {
+  num1: 'hotdog',
+  num2: 'elephant',
+  op: '*',
+}
+
+const exampleDivideByZero = {
+  num1: 5,
+  num2: 0,
+  op: 'divide',
+}
+
+/// CALCULATOR FUNCTIONALITY ///
+const operator = {
+  add: '+',
+  subtract: '-',
+  divide: '/',
+  multiply: '*',
+  exponent: '**',
+  modulo: '%',
+}
+
+function verifyNum(input) {
+  return isNaN(input) || !input
+}
+
+function printNum(data, result) {
```

Comment:
Great work encapsulating this functionality in a function to enhance user experience.